Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/NMDataServiceTest.cs

```diff
@@ -137,52 +137,134 @@ public void WhenGetDailyBatchDetailsFilesIsCalled_ThenShouldExecuteCatch()
+            A.CallTo(() => _fakefileShareService.FssSearchAttributeAsync("", CorrelationId)).Returns(res);
 
-            Assert.AreEqual(totalWeeks + 1, result.Count);
+            List<YearWeekModel> result = await _nMDataService.GetAllYearWeek(CorrelationId);
+            Assert.IsEmpty(result);
+            Assert.AreEqual(result.Count, ExpectedRecordCount);
         }
+
         [Test]
-        public void WhenGetAllWeeksofYearIsCalled_ThenForPastYearShouldReturnAllWeeksThatYear()
+        public async Task WhenGetAllYearWeekIsCalledwithValidTokenNoYearWeekdata_ThenShouldReturnNoList()
         {
-            DateTimeFormatInfo dateTimeFormatInfo = DateTimeFormatInfo.CurrentInfo;
+            int ExpectedRecordCount = 0;
+            A.CallTo(() => _fakeAuthFssTokenProvider.GenerateADAccessToken(A<string>.Ignored));
 
-            Calendar calender = dateTimeFormatInfo.Calendar;
+            IResult<BatchAttributesSearchResponse> res = SetAttributeSearchNoYearWeekData();
+            A.CallTo(() => _fakefileShareService.FssSearchAttributeAsync("", CorrelationId)).Returns(res);
 
-            int year = DateTime.Now.Year - 1;
+            List<YearWeekModel> result = await _nMDataService.GetAllYearWeek(CorrelationId);
 
-            DateTime lastdate = new(year, 12, 31);
+            Assert.IsEmpty(result);
+            Assert.AreEqual(result.Count, ExpectedRecordCount);
+        }
 
-            int totalWeeks = calender.GetWeekOfYear(lastdate, dateTimeFormatInfo.CalendarWeekRule, dateTimeFormatInfo.FirstDayOfWeek);
+        [Test]
+        public async Task WhenGetAllYearWeekIsCalledwithValidToken_ThenShouldReturnYearWeekList()
+        {
+            int ExpectedRecordCount = 3;
+            A.CallTo(() => _fakeAuthFssTokenProvider.GenerateADAccessToken(A<string>.Ignored));
+
+            IResult<BatchAttributesSearchResponse> res = SetAttributeSearchResult();
+            A.CallTo(() => _fakefileShareService.FssSearchAttributeAsync("", CorrelationId)).Returns(res);
 
-            List<KeyValuePair<string, string>> result = _nMDataService.GetAllWeeksOfYear(year, CorrelationId);
+            List<YearWeekModel> result = await _nMDataService.GetAllYearWeek(CorrelationId);
 
-            Assert.AreEqual(totalWeeks + 1, result.Count);
+            Assert.IsNotEmpty(result);
+            Assert.AreEqual(result.Count, ExpectedRecordCount);
         }
 
         [Test]
-        public void WhenGetAllYearsIsCalled_ThenShouldReturn4Records()
+        public async Task WhenGetWeeklyFilesResponseModelsAsyncIsCalled_ThenShouldReturnsShowWeeklyFilesResponseModelCount()
         {
-            const int yearsCount = 4;
-            List<KeyValuePair<string, string>> result = _nMDataService.GetAllYears(CorrelationId);
-            Assert.AreEqual(yearsCount, result.Count);
+            const int year = 2022, week = 15, expectedShowFilesResponseModelRecordCount = 4, expectedYearAndWeekRecordCount = 3;
```

Comment:
Please split these declarations onto separate lines:
```suggestion
            const int year = 2022;
            const int week = 15;
            const int expectedShowFilesResponseModelRecordCount = 4;
            const int expectedYearAndWeekRecordCount = 3;
```

Reply:
Declarations on separate lines implemented